payment_gateway/models/gateway_transaction.py

```diff
@@ -96,18 +94,21 @@ def _selection_capture_payment(self):
     used_3d_secure = fields.Boolean(
         help="Tic if this transaction have used 3d secure")
 
+    @api.multi
     @api.depends('origin_id')
     def _compute_origin(self):
         for record in self:
             record.res_id = record.origin_id.id
             record.res_model = record.origin_id._name
 
+    @api.multi
     def _get_amount_to_capture(self):
-        origin = self.origin_id
-        if origin._name == 'account.invoice':
-            return origin.residual
-        elif origin._name == 'sale.order':
-            return origin.amount_total
+        """
+        Return the amount to capture depending on the target model
+        :return: float
+        """
+        self.ensure_one()
+        return self.origin_id._get_transaction_to_capture_amount()
```

Comment:
Hello, overall I like the refactor. However I see a possible side effect here as currently the code is able to deal with capturing amounts on invoices (using the residual value) while the new code don't deal with this anymore. @sebastienbeau I think you need to check if we have downstream modules impacted that will need an adaptation may be. I'm speaking about this piece of code removal:
```
if origin._name == 'account.invoice':
            return origin.residual
```

Reply:
Hi @rvalyi , I'll put back this amount field for `account.invoice` (using new mixin abstract model)
I removed it because I didn't see any usage of it.

Reply:
I'm not sure if there is a real use case for it (past this code snippet). @sebastienbeau can you tell us?

Reply:
There is not real usecase for now so we can remove it